sympy/solvers/inequalities.py

```diff
@@ -982,3 +986,482 @@ def reduce_inequalities(inequalities, symbols=[]):
+    for i, var in enumerate(S):
+        v = var[0]
+        n = int(var[2])
+        if v == "y":
+            argmin_dual[n] = 0
+        else:
+            argmax[n] = M[i, -1]
+
+    return M[-1, -1], argmax, argmin_dual
+
+def linprog_from_matrices(A, B, C):
+    """
+    Because of the duality of linear programming, there are two valid ways to
+    interpret this function. The first is as a solver for the standard
+    maximization problem:
+        Maximizing Cx constrained to Ax <= B and x >= 0.
+    The second as is a solver for the standard minimization problem:
+        Minimizing y^{T}B constrained to y^{T}A >= C^{T} and y >= 0.
+
+    x is a column vector of variables, and y is a column vector of dual
+    variables.
+
+    If we keep the maximization interpretation in mind, then matrix A and
+    column vector B represent a set of constraints. For example,
+        1*x + 0*y <= 2
+        0*x + 0*y <= 4
+    can be represented as:
+    >>> from sympy.matrices.dense import Matrix
+    >>> A = Matrix([[1, 0],
+    ...             [0, 1]])
+    >>> B = Matrix([[2],
+    ...             [4]])
+
+    In this interpretation, row vector C is an objective function to be
+    maximized. For example,
+        3x + y
+    can be represented as:
+    >>> C = Matrix([[3, 1]])
+
+    Parameters
+    ==========
+
+    Under the maximization interpretation, m is the number of constraints and n
+    is the number of variables.
+
+    A : Matrix with shape (m, n)
+        Must contain only Rational and float coefficients.
+
+    B : Matrix with shape (m, 1)
+        Must contain only Rational and float coefficients.
+
+    C: Matrix with shape (1, n)
+        Must contain only Rational and float coefficients.
+
+    Returns
+    =======
+
+    Returns a triplet of solutions optimum, argmax, and argmax_dual:
+
+    optimum : float or Rational
+        maximum value of objective function under the constraints
+
+    argmax : list of floats and/or Rationals
+        x
+
+    argmax_dual : list of floats and/or Rationals
+        y
+
+    Examples
+    ========
+
+    Suppose we want to maximize
+        3x+y
+    subject to
+        x+y <= 2
+
+    >>> from sympy.matrices.dense import Matrix
+    >>> from sympy.solvers.inequalities import linprog_from_matrices
+    >>> A = Matrix([[1, 1]])
+    >>> B = Matrix([[2]])
+    >>> C = Matrix([[3, 1]])
+    >>> optimum, argmax, argmax_dual = linprog_from_matrices(A, B, C)
+    >>> optimum
+    6
+    >>> argmax
+    [2, 0]
+    >>> argmax_dual
+    [3]
+
+    Suppose we want to maximize
+        3x+y
+    subject to
+        x <= 2
+        y <= 4
+
+    >>> A = Matrix([[1,0], [0, 1]])
+    >>> B = Matrix([[2],[4]])
+    >>> C = Matrix([[3, 1]])
+    >>> optimum, argmax, argmax_dual = linprog_from_matrices(A, B, C)
+    >>> optimum
+    10
+    >>> argmax
+    [2, 4]
+    >>> argmax_dual
+    [3, 1]
+
+    See Also
+    ========
+
+    linprog_maximize
+    find_feasible
+    """
+    return _simplex(A, B, C)
```

Comment:
Perhaps as a courtesy we could have `return _simplex(*[Matrix(i) for i in (A, B, C)])` so one doesn't have to explicity say `Matrix([[...]])`.